test/flycheck-test.el

```diff
@@ -4273,12 +4273,25 @@ The manifest path is relative to
    '(2 nil warning "in procedure call to `g1', expected a value of type `(procedure (* *) *)' but was given a value of type `number'"
        :checker scheme-chicken)))
 
+(flycheck-ert-def-checker-test scheme-chicken scheme io-type-error
+  (flycheck-ert-should-syntax-check
+   "language/chicken/io-type-warning.scm" 'flycheck/chicken-mode
+   '(4 nil warning "in procedure call to `read-all', expected argument #1 of type `(or input-port string)' but was given an argument of type `output-port'"
+       :checker scheme-chicken)))
+
 (flycheck-ert-def-checker-test scheme-chicken scheme syntax-error
   (flycheck-ert-should-syntax-check
    "language/chicken/syntax-error.scm" 'flycheck/chicken-mode
    '(1 nil error "not enough arguments\n\n\t(define)\n\n\tExpansion history:\n\n\textras.scm:630: loop\t  \n\textras.scm:633: get-output-string\t  \n\tlibrary.scm:3655: ##sys#substring\t  \n\tlibrary.scm:603: ##sys#make-string\t  \n\tlibrary.scm:511: ##sys#allocate-vector\t  \n\textras.scm:633: ##sys#print\t  \n\tlibrary.scm:3188: case-sensitive\t  \n\tlibrary.scm:3189: keyword-style\t  \n\tlibrary.scm:3190: ##sys#print-length-limit\t  \n\tlibrary.scm:3319: ##sys#number?\t  \n\tlibrary.scm:3349: outstr\t  \n\tlibrary.scm:3188: g5138\t  \n\tsupport.scm:122: print-call-chain\t  \n\tlibrary.scm:3882: ##sys#get-call-chain\t  \n\tlibrary.scm:3834: ##sys#make-vector\t  \n\tlibrary.scm:1371: ##sys#allocate-vector\t  \t<--"
        :checker scheme-chicken)))
 
+(flycheck-ert-def-checker-test scheme-chicken scheme syntax-error-no-line-number
+  (flycheck-ert-should-syntax-check
+   "language/chicken/syntax-error-no-line-number.scm" 'flycheck/chicken-mode
+   ;; As the chicken-compiler doesn't print a line number for these errors,
```

Comment:
But we probably do want the syntax error to be reported in the Flycheck buffer, lest we think everything is okay.

Since chicken does not report any line number, we can use a default one.  Please add an `:error-filter` to the checker definition like this:

```emacs-lisp
  :error-filter flycheck-fill-empty-line-numbers
```

This will add a default line number.

Reply:
That's a much better experience! Thanks for pointing this out.
I've added that line, reworked the message capture for the matching error pattern and updated the relevant test accordingly.
Please take another look.